src/utils/dispute-utils.js

```diff
@@ -222,74 +290,118 @@ function getRoundPhasesAndTime(courtConfig, round, currentPhase) {
+  const {
+    draftTermId,
+    delayedTerms,
+    number: roundId,
+    createdAt,
+    vote,
+    appeal,
+  } = round
+  const isCurrentRound = roundId === currentPhase.roundId
+  const { winningOutcome } = vote || {}
+
+  const disputeDraftStartTime = getTermStartTime(draftTermId, courtConfig)
+
+  // Case where we are in a next round and has not yet started
+  if (isCurrentRound && currentPhase.phase === DisputesTypes.Phase.NotStarted) {
+    return [
+      {
+        phase: DisputesTypes.Phase.NotStarted,
+        endTime: disputeDraftStartTime,
+        roundId,
+        active: true,
+      },
+    ]
+  }
 
-  const disputeDraftTermTime = getTermStartTime(
-    parseInt(draftTermId) + parseInt(delayedTerms),
-    courtConfig
-  )
+  const disputeDraftTermEndTime =
+    disputeDraftStartTime + delayedTerms * termDuration
+
+  const revealEndTime =
+    disputeDraftTermEndTime + termDuration * (commitTerms + revealTerms)
+
+  const appealEndTime = revealEndTime + termDuration * appealTerms
+  const confirmAppealEndTime =
+    appealEndTime + termDuration * appealConfirmationTerms
+
+  const roundAppealed = !!appeal
 
   const roundPhasesAndTime = [
     {
       phase: DisputesTypes.Phase.JuryDrafting,
       endTime: createdAt + termDuration * juryDraftingTerms,
-      active: DisputesTypes.Phase.JuryDrafting === currentPhase.phase,
-      roundId: roundId,
+      active:
+        isCurrentRound &&
+        DisputesTypes.Phase.JuryDrafting === currentPhase.phase,
+      roundId,
     },
     {
       phase: DisputesTypes.Phase.VotingPeriod,
-      endTime: disputeDraftTermTime + termDuration * commitTerms,
-      active: DisputesTypes.Phase.VotingPeriod === currentPhase.phase,
-      roundId: roundId,
+      endTime: disputeDraftTermEndTime + termDuration * commitTerms,
+      active:
+        isCurrentRound &&
+        DisputesTypes.Phase.VotingPeriod === currentPhase.phase,
+      roundId,
     },
     {
       phase: DisputesTypes.Phase.RevealVote,
-      endTime:
-        disputeDraftTermTime + termDuration * (commitTerms + revealTerms),
-      active: DisputesTypes.Phase.RevealVote === currentPhase.phase,
-      roundId: roundId,
+      endTime: revealEndTime,
+      active:
+        isCurrentRound && DisputesTypes.Phase.RevealVote === currentPhase.phase,
+      roundId,
+      outcome: winningOutcome,
+      showOutcome: dayjs(new Date()).isAfter(revealEndTime),
     },
     {
       phase: DisputesTypes.Phase.AppealRuling,
-      endTime:
-        disputeDraftTermTime +
-        termDuration * (commitTerms + revealTerms + appealTerms),
-      active: DisputesTypes.Phase.AppealRuling === currentPhase.phase,
-      roundId: roundId,
+      endTime: appealEndTime,
+      active:
+        isCurrentRound &&
+        DisputesTypes.Phase.AppealRuling === currentPhase.phase,
+      roundId,
+      outcome: roundAppealed ? appeal.appealedRuling : null,
+      showOutcome:
+        dayjs(new Date()).isAfter(appealEndTime) ||
```

Comment:
Should we store `dayjs(new Date())` above?

Reply:
Yup, good idea